include/RAJA/LegacyCompatibility.hxx

```diff
@@ -3,7 +3,7 @@
 
 #include "RAJA/internal/defines.hxx"
 
-#if !defined(__INTEL_COMPILER)
+#if (!defined(__INTEL_COMPILER)) && (!defined(RAJA_COMPILER_MSVC))
```

Comment:
RAJA_COMPILER_ICC is a build-time thing, but __INTEL_COMPILER is defined when the Intel compiler is used. I think that the latter is safer because it accounts for people mixing compilers.